libraries/resource_provider.rb

```diff
@@ -229,14 +229,23 @@ def recursive_diff(data, desired_data, str = '', indent = '')
     # Generic method to retrieve and return a resource from different resources using the type and name of the resource
     # Note: The resource class base API module is the one used by the current provider
     # @param [String, Symbol] resource_class_type Type of resource to be retrieved. e.g., :GoldenImage, :FCNetwork
-    # @param [String] resource_id Name for this resource. 'Resource1'
+    # @param [Hash, String, NilClass] resource_id data containing unique IDs for this resource. e.g. uri: 'rest/fake/123456ABCDEF'
+    #   If a String is passed, it assumes the attribute is the `name` for this resource. e.g. 'Resource1' is interpreted as `name: 'Resource1'`.
+    #   If nothing or nil is passed, the method returns nil since there is nothing to be retrieved.
```

Comment:
I like it, we can just pass in the contexts without worrying if they have actually been filled usually. For most cases I guess this would be best.